docs/v2-migration-guide.rst

```diff
@@ -179,6 +179,78 @@ bug fixes will be shipped to the 1.26.x release stream**.
 If your organization relies on urllib3 and is interested in continuing support you can learn
 more about the `Tidelift Subscription for Enterprise <https://tidelift.com/subscription/pkg/pypi-urllib3?utm_source=pypi-urllib3&utm_medium=referral&utm_campaign=docs>`_.
 
+**🤔 Common upgrading issues**
+-------------------------------
+
+The following sections explain common issues and their remediation. If all else fails, you should
+pin urllib3 to 1.26.x using ``urllib3<2`` in your requirements. For more details on the recommended
+way to handle your dependencies in general, see `Semantic Versioning Will Not Save You
+<https://hynek.me/articles/semver-will-not-save-you/>`_. The second half even uses urllib3 2.0 as an
+example.
+
+ssl module is compiled with OpenSSL 1.0.2.k-fips
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+  ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with OpenSSL 1.0.2k-fips  26 Jan 2017.
+  See: https://github.com/urllib3/urllib3/issues/2168
+
+Remediation depends on your system.
+
+- **AWS Lambda**: Upgrade to the Python 3.9 or 3.10 runtimes that come OpenSSL 1.1.1.
+  Alternatively, you can use a `custom Docker image
+  <https://aws.amazon.com/blogs/aws/new-for-aws-lambda-container-image-support/>`_ and ensure you
+  use a Python build that uses OpenSSL 1.1.1 or later.
+- **Amazon Linux 2**: Upgrade to `Amazon Linux 2023
+  <https://aws.amazon.com/linux/amazon-linux-2023/>`_. Alternatively, you can install OpenSSL 1.1.1
+  on Amazon Linux 2 using ``yum install openssl11 openssl11-devel`` and then install Python with a
+  tool like pyenv.
+- **Red Hat Enterpritse Linux 7 (RHEL 7)**: Upgrade to RHEL 8 or RHEL 9.
+- **Read the Docs**: Upgrade your `configuration file to use Ubuntu 22.04
```

Comment:
Let's give a concrete example with the config file here (ie `build.os: ubuntu-22.04`)

Reply:
Done in [`008c89e` (#3018)](https://github.com/urllib3/urllib3/pull/3018/commits/008c89eee4e2dfc55a2e3e729372cc7c3509fd9b)